src/components/PopupWithForm.js

```diff
@@ -18,46 +22,38 @@ class PopupWithForm extends Popup {
     }, {});
     return userInput;
   }
-  // Comment to the reviewer.
-  // I did not understand the logics of making a method. Could you please explain it?
 
-  // getForm() {
-  // return this._popupForm
-  // }
-  // Also, I did not understand your comment about setting input values using a method.
-  // I get them using getInputValues and then they are passed to the SEL method on line 39
-  // and on 40 it is executed (with the help of callback in index.js)
-  //
-  // //  setInputValues(data) {
-  //   this._inputList.forEach((input) => {
-  //     // Here you insert the `value` by the `name` of the input
-  //     input.value = data[input.name];
-  //   });
-  // }
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._defaultButtonText;
+    }
+  }
 
-  setEventListeners() {
+  // User input values setter
+  setInputValues(data) {
+    this._inputList.forEach((input) => {
+      data[input.name] = input.value;
+    });
+  }
+
+  // Event Listeners setter and submit logics
+  _setEventListeners() {
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       const inputValues = this._getInputValues();
       this._handleFormSubmit(inputValues);
       this._popupForm.reset();
+      this.close();
```

Comment:
You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

Reply:
done

Reply:
Done. 
Transferred the logics into then block in the PopupWithForm in index.js

```js
        cardsList.addItem(newUserCard); // здесь контейнер в статусе undefined, потому что он не в глобальном скоупе
        formValidators["new-card-form"].disableButton();
        newCardPopup.getForm().reset();
        newCardPopup.close();
        
      ```